pymysql/connections.py

```diff
@@ -641,6 +633,31 @@ def _config(key, arg):
             port = int(_config("port", port))
             bind_address = _config("bind-address", bind_address)
             charset = _config("default-character-set", charset)
+            if not ssl:
+                ssl = {}
+            ssl_config_params = {
+                "ca": _config("ssl-ca", ssl.get("ca")),
+                "capath": _config("ssl-capath", ssl.get("capath")),
+                "cert": _config("ssl-cert", ssl.get("cert")),
+                "key": _config("ssl-key", ssl.get("key")),
+                "cipher": _config("ssl-cipher", ssl.get("cipher")),
+            }
+            # Update parameters, but parameters coming from the class parameter
+            # must have preference.
+            for key, value in ssl_config_params.items():
+                ssl.setdefault(key, value)
+            # No SSL parameters were given either via the class parameter or
+            # the configuration file
+            if not any(x for x in ssl.values()):
+                ssl = None
```

Comment:
`if not any(ssl.values())`

And `read_default_file` option changes how `ssl = {"ca": ""}` is considered.
if `read_default_file=False`, ssl is overwritten to None.  Otherwise, passed `ssl` is used as-is.
I don't like it.